pkg/integrations/squid_exporter/squid_exporter.go

```diff
@@ -0,0 +1,111 @@
+package squid_exporter
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	se "github.com/boynux/squid-exporter/collector"
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/prometheus/client_golang/prometheus"
+
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+)
+
+// DefaultConfig is the default config for the squid integration
+var DefaultConfig = Config{
+	Address:  "localhost:3128",
+	Username: "",
+	Password: "",
+}
+
+var (
+	errNoAddress  = errors.New("no address was provided")
+	errNoHostname = errors.New("no hostname in provided address")
+	errNoPort     = errors.New("no port in provided address")
+)
+
+// Config is the configuration for the squid integration
+type Config struct {
+	Address  string `yaml:"address"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+	Host     string
+	Port     int
+}
+
+func (c *Config) validate() error {
+	if c.Address == "" {
+		return errNoAddress
+	}
+
+	host, scheme, err := net.SplitHostPort(c.Address)
+	if err != nil {
+		return err
+	}
+
+	if host == "" {
+		return errNoHostname
+	}
+	c.Host = host
+
+	if scheme == "" {
+		return errNoPort
+	}
+
+	port, _ := strconv.Atoi(scheme)
+	c.Port = port
+
+	return nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Config
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultConfig
+
+	type plain Config
+	return unmarshal((*plain)(c))
+}
+
+// Name returns the integration name this config is associated with.
+func (c *Config) Name() string {
+	return "squid"
+}
+
+// InstanceKey returns the addr of the squid instance.
+func (c *Config) InstanceKey(agentKey string) (string, error) {
+	return c.Address, nil
+}
+
+// NewIntegration returns the Squid Exporter Integration
+func (c *Config) NewIntegration(_ log.Logger) (integrations.Integration, error) {
+	return New(c)
```

Comment:
I notice that we still aren't passing in a logger here, I just want to note that any errors that the exporter here would not get viewed in the agent log. This is definitely undesireable but I'm not sure if it warrants blocking this PR. I think the agent team can determine if this is a blocker. 

Do not resolve this before getting their review :)

Reply:
I guess I just wanted to note that this component as currently written appears to swallow errors and the Grafana team should note that before accepting this component.